packages/aws-cdk-lib/core/test/annotations.test.ts

```diff
@@ -79,18 +79,87 @@ describe('annotations', () => {
     const app = new App();
     const stack = new Stack(app, 'S1');
     const c1 = new Construct(stack, 'C1');
-    Annotations.of(c1).addWarning('You should know this!');
-    Annotations.of(c1).addWarning('You should know this!');
-    Annotations.of(c1).addWarning('You should know this!');
-    Annotations.of(c1).addWarning('You should know this, too!');
+    Annotations.of(c1).addWarningV2('warning1', 'You should know this!');
+    Annotations.of(c1).addWarningV2('warning1', 'You should know this!');
+    Annotations.of(c1).addWarningV2('warning1', 'You should know this!');
+    Annotations.of(c1).addWarningV2('warning2', 'You should know this, too!');
     expect(getWarnings(app.synth())).toEqual([{
       path: '/S1/C1',
-      message: 'You should know this!',
+      message: 'warning1: You should know this!',
     },
     {
       path: '/S1/C1',
-      message: 'You should know this, too!',
+      message: 'warning2: You should know this, too!',
     }],
     );
   });
+
+  test('acknowledgeWarning removes warning', () => {
+    // GIVEN
+    const app = new App();
+    const stack = new Stack(app, 'S1');
+    const c1 = new Construct(stack, 'C1');
+
+    // WHEN
+    Annotations.of(c1).addWarningV2('MESSAGE1', 'You should know this!');
+    Annotations.of(c1).addWarningV2('MESSAGE2', 'You Should know this too!');
+    Annotations.of(c1).acknowledgeWarning('MESSAGE2', 'I Ack this');
+
+    // THEN
+    const assembly = app.synth();
+    let acknoledgement: any = {};
+    for (const s of Object.values(assembly.manifest.artifacts ?? {})) {
+      for (const [_path, md] of Object.entries(s.metadata ?? {})) {
+        for (const x of md) {
+          if (x.type === 'aws:cdk:acknowledge') {
+            acknoledgement.message = x.data as string;
+          }
+        }
+      }
+    }
+    expect(acknoledgement).toEqual({
+      message: {
+        id: 'MESSAGE2',
+        scopes: ['S1/C1'],
+        message: 'I Ack this',
+      },
+    });
+    expect(getWarnings(assembly)).toEqual([{
+      path: '/S1/C1',
+      message: 'MESSAGE1: You should know this!',
+    }]);
+  });
+
+  test('acknowledgeWarning removes warning on children', () => {
+    // GIVEN
+    const app = new App();
+    const stack = new Stack(app, 'S1');
+    const c1 = new Construct(stack, 'C1');
+    const c2 = new Construct(c1, 'C2');
+
+    // WHEN
+    Annotations.of(c2).addWarningV2('MESSAGE2', 'You Should know this too!');
+    Annotations.of(c1).acknowledgeWarning('MESSAGE2', 'I Ack this');
+
+    // THEN
+    const assembly = app.synth();
+    let acknoledgement: any = {};
+    for (const s of Object.values(assembly.manifest.artifacts ?? {})) {
+      for (const [_path, md] of Object.entries(s.metadata ?? {})) {
+        for (const x of md) {
+          if (x.type === 'aws:cdk:acknowledge') {
+            acknoledgement.message = x.data as string;
```

Comment:
Mutations in the `THEN` part of a test?

Reply:
🤦‍♂️

Reply:
Just asking, if you need to do this work it probably wants to go into a reusable function that's in the core library?